lib/order.rb

```diff
@@ -7,12 +13,70 @@ def initialize(id, products)
       @products = products
     end
 
+    # gets the total plus tax of all of the products
     def total
-      # TODO: implement total
-    end
+      product_total = 0
+      subtotal = 0
+      @products.each_value do |prices|
+        subtotal += prices
+      end
+      product_total = (subtotal * 0.075).round(2)+ subtotal
+      return product_total
 
+    end
+    # Adds new product to @product array
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+
+      if @products.include?(product_name)
+
+        return false
+      else
+        @products[product_name]= product_price
+        return true
+      end
+      return products
+
+    end
+
+    def self.all
+      all_orders = []
+      all_orders_split = []
+      CSV.open(FILE_NAME, 'r').each do |file|
+        all_orders << file
+      end
+      # sets the order id
+      all_orders.each do |order|
+        id = order[0].to_i
+        # breaks the product string up
+        order_split = order[1].split(';')
+        # breaks the product string up again into key/value pairs
+        order_hash = {}
+        order_split.each do |product|
+          items = product.split(':')
+          # assigns product key/value pairs to a hash
+          product_key = items[0]
+          product_value = items[1].to_f
+          order_hash[product_key] = product_value
+        end
+        order =  Order.new(id, order_hash)
+        all_orders_split << order
+
+      end
+      return all_orders_split
+    end
+
+    def self.find(id)
+      Grocery::Order.all.each do |order|
```

Comment:
This isn't working.  It's good that you're using `Order.all`  to get the list, but instead of returning `order.products` it should just `return order`

If the loop finishes without finding an order it should return `nil`